pilot/pkg/bootstrap/configcontroller.go

```diff
@@ -107,19 +107,18 @@ func (s *Server) initConfigController(args *PilotArgs) error {
 			log.Warnf("Disabled ingress status syncer due to %v", err)
 		} else {
 			s.addTerminatingStartFunc(func(stop <-chan struct{}) error {
-				leaderelection.
-					NewLeaderElection(args.Namespace, args.PodName, leaderelection.IngressController, s.kubeClient.Kube()).
-					AddRunFunction(func(leaderStop <-chan struct{}) {
-						// Start informers again. This fixes the case where informers for namespace do not start,
-						// as we create them only after acquiring the leader lock
-						// Note: stop here should be the overall pilot stop, NOT the leader election stop. We are
-						// basically lazy loading the informer, if we stop it when we lose the lock we will never
-						// recreate it again.
-						s.kubeClient.RunAndWait(stop)
-						log.Infof("Starting ingress controller")
-						ingressSyncer.Run(leaderStop)
-					}).
-					Run(stop)
+				// Start informers again. This fixes the case where informers for namespace do not start,
+				// as we create them only after acquiring the leader lock
+				// Note: stop here should be the overall pilot stop, NOT the leader election stop. We are
+				// basically lazy loading the informer, if we stop it when we lose the lock we will never
+				// recreate it again.
+				s.kubeClient.RunAndWait(stop)
+				le := leaderelection.NewLeaderElection(args.Namespace, args.PodName, leaderelection.IngressController, s.kubeClient.Kube())
```

Comment:
Maybe in a separate PR, but can you add an env to control if leader election is even tried, for the various features that depend on this ? 

With multi-revision control it's probably not good if a canary wins the leader. We most likely want in some cases to have a specific revision ( most stable ) handle the singleton jobs.